### PR TITLE
ili9xxx: add list of models on top of the document. 

### PR DESCRIPTION
**Pull request by nielsnl68**
Pleasing the list of allowed display models on top of the page. As suggested on the discord #documention channel.

## Description:
add list of models on top of the document. 

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -609,7 +609,13 @@ Display Components
     ST7735, components/display/st7735, st7735.jpg
     ST7789V, components/display/st7789v, st7789v.jpg
     ST7920, components/display/st7920, st7920.jpg
-    ILI9xxx, components/display/ili9xxx, ili9341.jpg
```
Comment (jesserockz) [inline index.rst:612]: I dont think we should remove this entry as it is still something people might look for once actually using the code

Comment (nielsnl68) [reply index.rst:612]: 👍🏻

File: components/display/ili9xxx.rst [hunk 2]
```diff
@@ -3,11 +3,19 @@ ILI9xxx TFT LCD Series
...
+    :description: Instructions for setting up ILI9xxx like TFT LCD display drivers.
     :image: ili9341.jpg
 
 .. _ili9xxx:
 
+Models
+------
+With this display driver you can control the following displays:
+  - ``M5STACK``, ``TFT 2.4``, ``TFT 2.4R``, ``S3BOX``, ``S3BOX_LITE``
+  - ``ILI9341``, ``ILI9342``, ``ILI9481``, ``ILI9486``, ``ILI9488``, ``ST7796``
```
Comment (kbx81) [inline components/display/ili9xxx.rst]: Just tweaking formatting a bit and alphabetizing:

```suggestion
  - ILI9341
  - ILI9342
  - ILI9481
  - ILI9486
  - ILI9488
  - M5STACK
  - S3BOX
  - S3BOX_LITE
  - ST7796
  - TFT 2.4
  - TFT 2.4R
```

**Review by kbx81 — CHANGES_REQUESTED** [review]

File: index.rst [hunk 3]
```diff
@@ -610,6 +610,18 @@ Display Components
...
+    ILI9342, components/display/ili9xxx, ili9341.jpg
+    ILI9481, components/display/ili9xxx, ili9341.jpg 
+    ILI9486, components/display/ili9xxx, ili9341.jpg 
+    ILI9488, components/display/ili9xxx, ili9341.jpg 
+    ST7796, components/display/ili9xxx, ili9341.jpg
+    M5STACK, components/display/ili9xxx, ili9341.jpg
+    TFT 2.4, components/display/ili9xxx, ili9341.jpg
+    TFT 2.4R, components/display/ili9xxx, ili9341.jpg
+    S3BOX, components/display/ili9xxx, ili9341.jpg
+    S3BOX_LITE, components/display/ili9xxx, ili9341.jpg
```
Comment (nielsnl68) [inline index.rst:623]:
``` suggestions
```

remove those; not needed.

Comment (nielsnl68) [inline index.rst]:
```suggestion

```

remove those; not needed.

**Review by kbx81 — APPROVED** [review]

**Pull request merged**